### PR TITLE
Update balena-sdk to 15.51.1

### PR DESCRIPTION
**Pull request by nitishagar**
Update balena-sdk from 15.48.0 to 15.51.0

Change-type: patch
  
HQ: https://jel.ly.fish/e450713f-001c-4470-b571-71724aeb0869
See: https://www.flowdock.com/app/rulemotion/r-product/threads/DZt_LvQvMCOoN6p2EceSu6MG4MF
See: https://www.flowdock.com/app/rulemotion/resin-tech/threads/IPdnkCLH2G0o7DSlkSS9nRetA4f  
See: https://github.com/balena-io/balena-cli/pull/2294

### CODE REVIEW

**Review by pdcastro — COMMENTED** [review]
Looks good @nitishagar 👍   Just a minor thing: The commit and PR specify `change-type: minor`, but from what I can tell, no new functionality is provided to CLI users. It looks to me like it fits in the category of "chore" dependency updates with no impact to end users, also like bug fixes or security updates, which I would normally classify as a "patch" (rather than "minor") update. 
If there is actually some new feature provided to users, then the commit message could perhaps be amended to reflect it (while keeping the commit body as it is).

**Comment by klutchell** [comment]
I'm a little wary that the following unrelated upstream changes would also be included:
- https://github.com/balena-io/balena-sdk/pull/1120
- https://github.com/balena-io-modules/balena-request/pull/156

These were supposed to be done around the same time as some other fixes for https://github.com/balena-io/balena-cli/issues/2152. However, we haven't figured out those other fixes yet and the PR mentioned above actually masks the `Z_BUF_ERR` and appears to download the image successfully even though the final file is incomplete.

So while on it's own it seems like a good fix to have, it actually hides another issue that we haven't solved yet. I can revert balena-requests, or revert the change in both repos while moving them forward, but I worry we will start seeing incomplete file downloads unless we get to the root of the os download pipeline issues.

Thoughts on how to proceed and unblock this PR?

**Comment by pdcastro** [comment]
> [...] and the PR mentioned above actually masks the `Z_BUF_ERR` and appears to download the image successfully even though the final file is incomplete.
> [...] I can revert balena-requests, or revert the change in both repos while moving them forward [...]

Thanks for the insight @klutchell 👍  Instead of reverting those changes, could the changes be enabled/disabled through a [feature switch](https://en.wikipedia.org/wiki/Feature_toggle)? If we could think of no better option, perhaps even an environment variable that the CLI could set, starting with this PR, say `process.env.MASK_Z_BUF_ERR=false`, either globally or in selected commands like `balena os download`. (I don't mean an environment variable that users would see/set, I mean simply CLI code assigning to `process.env.VAR_NAME`.)

**Comment by nitishagar** [comment]
For the following bit:
> If there is actually some new feature provided to users, then the commit message could perhaps be amended to reflect it (while keeping the commit body as it is).

Among other changes, we are looking to enable addition of `provision-key` name as a param for CLI to consume. Ref: https://github.com/balena-io/balena-cli/pull/2294

I took the default message with the update script, but can detail it based on the above. Let me know.

**Comment by pdcastro** [comment]
> If there is actually some new feature provided to users [...]
>> Among other changes, we are looking to enable addition of `provision-key` name as a param for CLI to consume. Ref: #2294

@nitishagar, but this PR (just bumping `balena-sdk`) does not add `provision-key` name as a param. I understand that this PR does not make any difference to CLI users (other than possibly introducing a bug in `balena os download` as mentioned by Kyle). In other words, I understand that there isn't anything that CLI users can do with the CLI after this PR is merged, that they could not do before it was merged. So this PR would be `Change-type: patch`, and #2294 would be `Change-type: minor`.

>> I took the default message with the update script, but can detail it based on the above. Let me know.

If my understanding above is correct, then the commit message is OK as it is, but `Change-type` should be patch.

The reason why it matters (a little bit) is that the commit message and new semver version will be visible in the [Changelog](https://github.com/balena-io/balena-cli/blob/master/CHANGELOG.md) that CLI end users and support agents refer to in order to find out what's new, and to decide whether to update the CLI.

**Comment by klutchell** [comment]
I would like to get the following PRs merged before this one if possible:
- https://github.com/balena-io/balena-sdk/pull/1126 (and update this PR to use sdk `v15.51.1`)
- https://github.com/balena-io/balena-cli/pull/2347 (under test)

File: package.json
Comment (pdcastro) [inline package.json:210]: @nitishagar, Kyle's PRs have been merged and `balena-sdk` v15.51.1 is now available. Please select `balena-sdk` v15.51.1 here and change the commit `Change-type` to patch, then we can get this PR merged. Thanks! 🙏

**Review by pdcastro — APPROVED** [review]

**Pull request merged**
